name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/earnings/EarningsViewModel.java

```diff
@@ -242,14 +249,45 @@ private void calculate()
 
                 if (!predicate.test(t))
                     continue;
+                long value = 0;
+
+                switch (mode)
+                {
+                    case FEE:
+                        if (t.getType() == AccountTransaction.Type.FEES
+                                        || t.getType() == AccountTransaction.Type.FEES_REFUND)
+                            value -= t.getMonetaryAmount().getAmount();
+                        else
+                            value -= t.getUnitSum(Unit.Type.FEE, converter).getAmount();
+                        break;
+                    case TAX:
+                        if (t.getType() == AccountTransaction.Type.TAXES
+                                        || t.getType() == AccountTransaction.Type.TAX_REFUND)
+                            value -= t.getMonetaryAmount().getAmount();
+                        else
+                            value -= t.getUnitSum(Unit.Type.TAX, converter).getAmount();
+                        break;
+                    case EXPENSES:
+                        if (t.getType() == AccountTransaction.Type.FEES
+                                        || t.getType() == AccountTransaction.Type.FEES_REFUND
+                                        || t.getType() == AccountTransaction.Type.TAXES
+                                        || t.getType() == AccountTransaction.Type.TAX_REFUND)
+                            value -= t.getMonetaryAmount().getAmount();
+                        else
+                        {
+                            value -= t.getUnitSum(Unit.Type.FEE, converter).getAmount();
+                            value -= t.getUnitSum(Unit.Type.TAX, converter).getAmount();
+                        }
+                        break;
+                    default:
+                        value = t.getMonetaryAmount().getAmount();
```

Comment:
Die Buchungen müssen nicht zwingend in der selben Währung sein (die können von verschiedenen Konten stammen). Bevor die aufsummiert werden, muss man den CurrencyConverter anwenden, z.B. mit

```java
long value = t.getMonetaryAmount().with(converter.at(t.getDateTime())).getAmount();
```

Reply:
Obsolete

Reply:
Obsolete